notebooks/machine-learning-pipelines/intro-to-pipelines/aml-pipelines-publish-and-run-using-rest-endpoint.ipynb

```diff
@@ -0,0 +1,469 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Copyright (c) Microsoft Corporation. All rights reserved.  \n",
+    "Licensed under the MIT License."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![Impressions](https://PixelServer20190423114238.azurewebsites.net/api/impressions/MachineLearningNotebooks/how-to-use-azureml/machine-learning-pipelines/intro-to-pipelines/aml-pipelines-publish-and-run-using-rest-endpoint.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# How to Publish a Pipeline and Invoke the REST endpoint\n",
+    "In this notebook, we will see how we can publish a pipeline and then invoke the REST endpoint."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites and Azure Machine Learning Basics\n",
+    "If you are using an Azure Machine Learning Notebook VM, you are all set. Otherwise, make sure you go through the [configuration Notebook](https://aka.ms/pl-config) first if you haven't. This sets you up with a working config file that has information on your workspace, subscription id, etc. \n",
+    "\n",
+    "### Initialization Steps"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import azureml.core\n",
+    "from azureml.core import Workspace, Datastore, Experiment, Dataset\n",
+    "from azureml.data import OutputFileDatasetConfig\n",
+    "from azureml.core.compute import AmlCompute\n",
+    "from azureml.core.compute import ComputeTarget\n",
+    "\n",
+    "# Check core SDK version number\n",
+    "print(\"SDK version:\", azureml.core.VERSION)\n",
+    "\n",
+    "from azureml.pipeline.core import Pipeline\n",
+    "from azureml.pipeline.steps import PythonScriptStep\n",
+    "from azureml.pipeline.core.graph import PipelineParameter\n",
+    "\n",
+    "print(\"Pipeline SDK-specific imports completed\")\n",
+    "\n",
+    "ws = Workspace.from_config()\n",
+    "print(ws.name, ws.resource_group, ws.location, ws.subscription_id, sep = '\\n')\n",
+    "\n",
+    "# Default datastore (Azure blob storage)\n",
+    "# def_blob_store = ws.get_default_datastore()\n",
+    "def_blob_store = Datastore(ws, \"workspaceblobstore\")\n",
+    "print(\"Blobstore's name: {}\".format(def_blob_store.name))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Compute Targets\n",
+    "#### Retrieve an already attached  Azure Machine Learning Compute"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core.compute_target import ComputeTargetException\n",
+    "\n",
+    "aml_compute_target = \"cpu-cluster\"\n",
+    "try:\n",
```

Comment:
remove - see other examples - also no try/except for things like this

Reply:
I left try catch because AzureML documenttaion still seggests that method for creating compute: https://docs.microsoft.com/en-us/azure/machine-learning/how-to-create-manage-compute-instance?tabs=python
is that no longer the case?

Reply:
that is no longer the case

Reply:
is there an example ?

Reply:
of compute create? it's in the contributing guide - "don't create compute targets" - instead use one of the existing ones created in workspace setup script: https://github.com/Azure/azureml-examples/blob/main/setup-workspace.py

Reply:
I elminated Try and catch and set up If nstatements to check whether compute has already been created, based on the title of the compute created in the link above, it should catch "cpu-compute"/ "gpu-compute" and use whats already created.